vector/src/main/java/im/vector/activity/VectorHomeActivity.java

```diff
@@ -644,6 +651,34 @@ public void onClick(DialogInterface dialog, int which) {
         }
     }
 
+    /**
+     * Display a dialog to let the user chooses if he would like to use analytics tracking
+     */
+    private void promptForAnalyticsTracking() {
+
+        AlertDialog.Builder builder = new AlertDialog.Builder(this);
+        builder.setMessage(R.string.settings_opt_in_of_analytics_prompt);
+        builder.setPositiveButton(R.string.settings_opt_in_of_analytics_ok, new DialogInterface.OnClickListener() {
+                    @Override
+                    public void onClick(DialogInterface dialog, int which) {
+                        setAnalyticsAuthorization(true);
+                    }
+                })
+                .setNegativeButton(R.string.no, new DialogInterface.OnClickListener() {
+                    @Override
+                    public void onClick(DialogInterface dialog, int which) {
+                        setAnalyticsAuthorization(false);
+                    }
+                })
+                .create()
+                .show();
+    }
+
+    private void setAnalyticsAuthorization(boolean useAnalytics) {
+        PreferencesManager.setUseAnalytics(VectorHomeActivity.this, useAnalytics);
+        PreferencesManager.setAuthorizationToUseAnalyticsRequest(VectorHomeActivity.this);
```

Comment:
I think this setting is not a preference, but rather a SharedPref.

Reply:
Yes it is a SharedPref and it is managed like a SharedPref in PreferenceManager class.